include/openPMD/Datatype_internal.hpp

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include "openPMD/auxiliary/TypeTraits.hpp"
+
+#include <array>
+#include <cstddef>
+#include <vector>
+
+namespace openPMD
+{
+namespace detail
+{
+    template <typename DoDetermineDatatype>
+    struct BasicDatatype
+    {
+        using DT = typename DoDetermineDatatype::DT_enum;
+
+        template <typename T>
+        constexpr static DT call()
+        {
+            if constexpr (auxiliary::IsVector_v<T>)
+            {
+                return DoDetermineDatatype::template call<
+                    typename T::value_type>();
+            }
+            else if constexpr (auxiliary::IsArray_v<T>)
+            {
+                return DoDetermineDatatype::template call<
+                    typename T::value_type>();
+            }
+            else
+            {
+                return DoDetermineDatatype::template call<T>();
+            }
+#if defined(__INTEL_COMPILER)
+/*
+ * ICPC has trouble with if constexpr, thinking that return statements are
```

Comment:
I think NVCC 11.0 has a similar issue, potentially check and add a similar suppression?
Around 11.5+, the false warning is solved.